clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java

```diff
@@ -168,4 +169,23 @@ public long getUsedBytes() throws IOException {
       mContext.releaseMasterClient(blockMasterClient);
     }
   }
+
+  public void promote(long blockId) throws IOException {
+    BlockMasterClient blockMasterClient = mContext.acquireMasterClient();
+    try {
+      BlockInfo info = blockMasterClient.getBlockInfo(blockId);
+      if (info.getLocations().isEmpty()) {
+        throw new IOException("Block " + blockId + " is not available in Tachyon.");
```

Comment:
Oh when I say `not available in Tachyon` I mean no current worker has it so it cannot be promoted. Writing a block into Tachyon space is different from promotion I think.